iree/samples/rt/vulkan/vk_graphics_integration.cc

```diff
@@ -70,15 +72,102 @@ static void check_vk_result(VkResult err) {
 
 #define CHECK_IREE_OK(status) CHECK_EQ(IREE_STATUS_OK, (status))
 
-static void SetupVulkan(const char** extensions, uint32_t extensions_count) {
+static std::vector<const char*> GetInstanceExtensions(
+    SDL_Window* window, iree_hal_vulkan_extensibility_options_t options) {
+  // Ask SDL for its list of required instance extensions.
+  uint32_t sdl_extensions_count = 0;
+  SDL_Vulkan_GetInstanceExtensions(window, &sdl_extensions_count, NULL);
+  const char** sdl_extensions = new const char*[sdl_extensions_count];
+  SDL_Vulkan_GetInstanceExtensions(window, &sdl_extensions_count,
+                                   sdl_extensions);
+
+  // Ask IREE for its list of required instance extensions.
+  iree_host_size_t iree_required_extensions_count = 0;
+  iree_hal_vulkan_get_extensions(IREE_HAL_VULKAN_INSTANCE_REQUIRED, options, 0,
+                                 NULL, &iree_required_extensions_count);
+  const char** iree_required_extensions =
+      new const char*[iree_required_extensions_count];
+  iree_hal_vulkan_get_extensions(IREE_HAL_VULKAN_INSTANCE_REQUIRED, options,
+                                 iree_required_extensions_count,
+                                 iree_required_extensions,
+                                 &iree_required_extensions_count);
+
+  // Ask IREE for its list of optional instance extensions.
+  iree_host_size_t iree_optional_extensions_count = 0;
+  iree_hal_vulkan_get_extensions(IREE_HAL_VULKAN_INSTANCE_OPTIONAL, options, 0,
+                                 NULL, &iree_optional_extensions_count);
+  const char** iree_optional_extensions =
+      new const char*[iree_optional_extensions_count];
+  iree_hal_vulkan_get_extensions(IREE_HAL_VULKAN_INSTANCE_OPTIONAL, options,
+                                 iree_optional_extensions_count,
+                                 iree_optional_extensions,
+                                 &iree_optional_extensions_count);
+
+  // Merge extensions lists, including optional and required for simplicity.
+  std::set<const char*> ext_set;
+  for (int i = 0; i < sdl_extensions_count; ++i) {
+    ext_set.insert(sdl_extensions[i]);
+  }
+  for (int i = 0; i < iree_required_extensions_count; ++i) {
+    ext_set.insert(iree_required_extensions[i]);
+  }
+  for (int i = 0; i < iree_optional_extensions_count; ++i) {
+    ext_set.insert(iree_optional_extensions[i]);
+  }
+  std::vector<const char*> extensions(ext_set.begin(), ext_set.end());
+  return extensions;
+}
+
+static std::vector<const char*> GetDeviceExtensions(
+    iree_hal_vulkan_extensibility_options_t options) {
+  // Ask IREE for its list of required device extensions.
+  iree_host_size_t iree_required_extensions_count = 0;
+  iree_hal_vulkan_get_extensions(IREE_HAL_VULKAN_DEVICE_REQUIRED, options, 0,
+                                 NULL, &iree_required_extensions_count);
+  const char** iree_required_extensions =
```

Comment:
you can use std::vector<const char*> to take care of this and then wrap get_extensions; like:
```
std::vector<const char*> GetExtensions(args...) {
  iree_host_size_t required_count;
  .._get_extensions(&required_count);
  std::vector<const char*> extensions(required_count);
  .._get_extensions(extensions.data(), &required_count);
  return extensions;
}
```

then you won't be leaking memory :)

Reply:
Replaced `new const char*` use with `std::vector<const char*>` and added the helper.